application/src/battery.cpp

```diff
@@ -204,8 +211,14 @@ void Battery::updateData()
         healthFile->close();
     }
 
+    // e.g. PineTab outputs an integer in centi-centigrade
+    // Note that the formatter in the QML page, and the logger divide by 10 again!
+    float tempCorrectionFactor = 1.0 ;
+    if(temperatureFile->fileName().contains(QStringLiteral("xp20x-battery"))) {
+        tempCorrectionFactor = 10.0 ;
+    }
     if(temperatureFile && temperatureFile->open(QIODevice::ReadOnly)) {
-        nextTemperature = temperatureFile->readLine().trimmed().toInt();
+        nextTemperature = temperatureFile->readLine().trimmed().toInt() / tempCorrectionFactor;
```

Comment:
Making a string search every update seems avoidable. I'd prefer having `tempCorrectionFactor` in the class, and have it set in the constructor (around line 95?) instead.